src/gval/utils/loading_datasets.py

```diff
@@ -238,205 +236,81 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
+    catalog_df = pd.concat(item_dfs, ignore_index=True)
 
-    """
+    if attribute_allow_list is not None:
+        catalog_df = catalog_df[attribute_allow_list]
 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        # Call cataloging url, search, and convert to xarray
-        catalog = pystac_client.Client.open(url)
-
-        stac_items = catalog.search(
-            datetime=time,
-            collections=[collection],
-            max_items=max_items,
-            intersects=intersects,
-            bbox=bbox,
-            query=query,
-        ).get_all_items()
-
-        stack = stackstac.stack(stac_items, resolution=resolution)
-
-        # Only get unique time indices in case there are duplicates
-        _, idxs = np.unique(stack.coords["time"], return_index=True)
-        stack = stack[idxs]
-
-        # Aggregate if there is more than one time
-        if stack.coords["time"].shape[0] > 1:
-            crs = stack.rio.crs
-            if time_aggregate == "mean":
-                stack = stack.mean(dim="time")
-                stack.attrs["time_aggregate"] = "mean"
-            elif time_aggregate == "min":
-                stack = stack.min(dim="time")
-                stack.attrs["time_aggregate"] = "min"
-            elif time_aggregate == "max":
-                stack = stack.max(dim="time")
-                stack.attrs["time_aggregate"] = "max"
-            else:
-                raise ValueError("A valid aggregate must be used for time ranges")
-
-            stack.rio.write_crs(crs, inplace=True)
-        else:
-            stack = stack[0]
-            stack.attrs["time_aggregate"] = "none"
-
-        # Select specific bands
-        if bands is not None:
-            bands = [bands] if isinstance(bands, str) else bands
-            stack = stack.sel({"band": bands})
-
-        band_metadata = (
-            stack.coords["raster:bands"] if "raster:bands" in stack.coords else None
-        )
-        if "band" in stack.dims:
-            og_names = [name for name in stack.coords["band"]]
-            names = [f"band_{x + 1}" for x in range(len(stack.coords["band"]))]
-            stack = stack.assign_coords({"band": names}).to_dataset(dim="band")
-
-            for metadata, var, og_var in zip(band_metadata, stack.data_vars, og_names):
-                _set_nodata(stack[var], metadata, nodata_fill)
-                stack[var] = _set_crs(stack[var], band_metadata)
-                stack[var].attrs["original_name"] = og_var
-
-        else:
-            stack = stack.to_dataset(name="band_1")
-            _set_nodata(stack["band_1"], band_metadata, nodata_fill)
-            stack["band_1"] = _set_crs(stack["band_1"])
-            stack["band_1"].attrs["original_name"] = (
-                bands[0] if isinstance(bands, list) else bands
-            )
+    if attribute_block_list is not None:
+        catalog_df = catalog_df.drop(attribute_block_list, axis=1)
```

Comment:
What if both lists are None? Mutually exclusive arguments?

Reply:
If both lists are none it is fine, I added logic to forbid a column being in both lists.